### PR TITLE
Upgraded projects and NuGet packages to be compliant with .net core 3.1

### PR DESCRIPTION
**Pull request by chrisbbe**
- All `Microsoft.*` packages upgraded to latest 3.* version.
- All Google NuGet packages upgraded to latest version.
- Test projects now targets `netcoreapp3.1`

### CODE REVIEW

**Review by jskeet — APPROVED** [review]
This looks good to me, but I'd like @amanda-tarafa to take a look too before we merge.

(There are various aspects of CI we should really get set up on this repo too.)

File: Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.IntegrationTests/Google.Cloud.AspNetCore.IntegrationTests.csproj [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.1</TargetFrameworks>
     <LangVersion>latest</LangVersion>
     <IsPackable>false</IsPackable>
     <AssemblyOriginatorKeyFile>../../GoogleApis.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <NoWarn>1701;1702;1705;xUnit2004;xUnit2013;AD0001</NoWarn>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (amanda-tarafa) [inline Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.IntegrationTests/Google.Cloud.AspNetCore.IntegrationTests.csproj:10]:
```suggestion
```

File: Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.IntegrationTests/Google.Cloud.AspNetCore.IntegrationTests.csproj [hunk 2]
```diff
@@ -1,23 +1,26 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
```
Comment (amanda-tarafa) [inline Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.IntegrationTests/Google.Cloud.AspNetCore.IntegrationTests.csproj:3]: Just leave it as the first one, as it's faster to find, and how it is in all our other projects. Thanks
```suggestion
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
```

File: Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.Tests/Google.Cloud.AspNetCore.Tests.csproj [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.1</TargetFrameworks>
     <LangVersion>latest</LangVersion>
     <IsPackable>false</IsPackable>
     <AssemblyOriginatorKeyFile>../../GoogleApis.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <NoWarn>1701;1702;1705;xUnit2004;xUnit2013;AD0001</NoWarn>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (amanda-tarafa) [inline Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.Tests/Google.Cloud.AspNetCore.Tests.csproj:10]: Same here, move it to the top.

**Review by amanda-tarafa — COMMENTED** [review]
Looks good, just made a couple of suggestions.

File: Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.Tests/Google.Cloud.AspNetCore.Tests.csproj (hunk 3 above)
Comment (chrisbbe) [reply Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.Tests/Google.Cloud.AspNetCore.Tests.csproj]: Done

**Comment by jskeet** [comment]
Closing this PR now as I believe it's obsolete.

**Pull request closed without merging**